include/bluetooth/mesh/main.h

```diff
@@ -93,6 +120,21 @@ struct bt_mesh_prov {
 	/** Supported Input OOB Actions */
 	uint16_t       input_actions;
 
+	/** @brief Provisioning Capabilities.
+	 *
+	 *  This callback notifies the application that the provisioning capabilities
+	 *  of the unprovisioned device has been received.
+	 *
+	 *  The application can consequently call bt_mesh_auth_method_set_<*> to
+	 *  select suitable provisioning oob authentication method.
+	 *
+	 *  When this callback returns, the provisioner will start authentication with
+	 *  the chosen method.
+	 *
+	 *  @param cap capabilities supported by device.
+	 */
+	void         (*capabilities)(const struct bt_mesh_prov_capabilities *cap);
```

Comment:
I realise this has already been happening with other `bt_mesh_prov` struct members, but it surprises me that there are no preprocessor conditionals here to include some members only if provisioner support has been enabled.

Reply:
you mean change as:

``` C

#if defined(CONFIG_BT_MESH_PROVISIONER)
	void         (*capabilities)(const struct bt_mesh_prov_capabilities *cap);
#endif
```

Reply:
Yes, except it'd probably make sense to group all of them so that you have a single `#if - #endif` pair. Anyway, as I said, this is not new, so this kind of stuff (if it makes sense) can be done in a later PR

Reply:
Alternatively the provisioner-specific members could be moved out to a separate struct that's registered independently.

Reply:
+1 on separating them to a different struct, the application typically wouldn't implement both roles, and if they do, I'd expect they'd be handled by two separate modules. This would also allow proper documentation of the new usage of the OOB auth callbacks

Reply:
This is a missing problem, which is worth solving in the future PR, or @trond-snekvik  can you amend these changes to this PR.

Reply:
Let's do it in a separate PR